corehq/apps/locations/models.py

```diff
@@ -424,6 +425,7 @@ def delete(self, *args, **kwargs):
 
     full_delete = delete
 
+    @quickcache(['self.location_id', 'include_self'], timeout=30 * 60)
```

Comment:
I don't think this is a great target for `quickcache`, as it returns an unevaluated queryset.  Callers do things like `location.get_descendants().exists()` or `location.get_descendants().filter(is_archived=True)`, and I'm not sure exactly how that'd behave with a cache here.